### PR TITLE
More documentation

### PR DESCRIPTION
**Pull request by dolgopolovwork**

### CODE REVIEW

File: btc-withdrawal/README.md [hunk 1]
```diff
@@ -1,33 +1,26 @@
...
-* `btc-withdrawal.btcConsensusCredential` - this account is used to create 'withdrawal consensus'. It creates other accounts named after withdrawal operation `hashCode()` in `btcConsensus` domain and saves consensus information in it.
+* `btc-withdrawal.btcConsensusCredential` - this account is used to create 'withdrawal consensus'.
 * `btc-withdrawal.registrationCredential` - the account that was used to register D3 clients in the Bitcoin network. This account is needed to get clients whitelists.
-* `btc-withdrawal.changeAddressesStorageAccount` - Iroha account that we use to store Bitcoin change addresses. Change addresses are created automatically. 
-* `btc-withdrawal.irohaBlockQueue` - RabbitMQ queue name that `chain-adapter` uses to deliver Iroha blocks
+* `btc-withdrawal.changeAddressesStorageAccount` - Iroha account that we use to store Bitcoin change addresses.
+* `btc-withdrawal.irohaBlockQueue` - name of a RabbitMQ queue that will be used to read Iroha blocks from
```
Comment (Mingela) [inline btc-withdrawal/README.md]:
```suggestion
* `btc-withdrawal.irohaBlockQueue` - name of the RabbitMQ queue to read Iroha blocks from
```

File: README.md [hunk 2]
```diff
@@ -1,11 +1,67 @@
...
+* Redeem Script - a special script that is used to describe the terms of locking/releasing coins.
+
+* RegTest - local Bitcoin testing network
+
+* TestNet - public Bitcoin testing network
+
+* MainNet - public Bitcoin main network
+
+## Bitcoin network
+Notary communicates with two entities: ``BitcoinD`` and `IrohaD`. The communication between `BitcoinD` and Notary goes through SPV node. By doing so, we can be sure that Notary won't run out of disk space while working with Bitcoin.
```
Comment (Mingela) [inline README.md]:
```suggestion
Notary communicates with two entities: ``BitcoinD`` and `IrohaD`. The communication between `BitcoinD` and Notary goes through SPV node. By doing so, we ensure Notary won't run out of disk space while working with Bitcoin. 
```

File: README.md [hunk 3]
```diff
@@ -1,11 +1,67 @@
...
+
+2. Every Notary creates a MultiSig address using public keys generated by all the Notaries.
+
+#### Implementation details/issues 
+Address generation is a time-consuming process. It may take approximately 10-15 seconds to collect all the keys. To speed up the registration process, we generate MultiSig addresses up front.
+
+[More details](https://github.com/d3ledger/d3-btc/blob/develop/btc-address-generation/README.md)
+
+## Client Registration 
+This event will assign a Bitcoin MultiSig address to recently created Iroha account.
```
Comment (Mingela) [inline README.md]:
```suggestion
This event assigns a Bitcoin MultiSig address to the recently created Iroha account. 
```

File: README.md [hunk 4]
```diff
@@ -1,11 +1,67 @@
...
+
+## Client Registration 
+This event will assign a Bitcoin MultiSig address to recently created Iroha account. 
+
+Client sends a registration request to particular Notary node. 
+Notary chooses MultiSig address and creates a transaction in Iroha to create a "client to Bitcoin address" relationship.  
+
+
+## Deposit Service
+Each Notary watches for transactions of MultiSig addresses assigned to our clients in the Bitcoin network. Whenever we see a transaction with coins being sent to one of the addresses, Notaries increase its balance.
```
Comment (Mingela) [inline README.md]:
```suggestion
Each Notary listens to transactions of the MultiSig addresses assigned to our clients in the Bitcoin network. Whenever we see a transaction with coins being sent to one of the addresses, Notaries increase its balance.
```

File: btc-address-generation/README.md [hunk 5]
```diff
@@ -1,26 +1,38 @@
...
 * `btc-address-generation.changeAddressesStorageAccount` -  Iroha account that is responsible for change addresses storage. Addresses are stored in this account details.
 * `btc-address-generation.healthCheckPort` - port of health check endpoint. A health check is available on `http://host:healthCheckPort/actuator/health`. This service checks if `btc-address-generation` is able to listen to Iroha blocks.
 * `btc-address-generation.threshold` - a number of MultiSig addresses that must be created in advance.
-* `btc-address-generation.nodeId` - identifier of Notary node. This identifier must correlate to an identifier that is set in `btc-registration` configuration file on the same node. This value must be different on different nodes. We use this identifier as a mechanism of synchronization issues resistance.
+* `btc-address-generation.nodeId` - identifier of node. This identifier must correlate to an identifier that is set in `btc-registration` configuration file on the same node. This value must be different on different nodes.
+* `btc-address-generation.irohaBlockQueue` - name of a RabbitMQ queue that will be used to read Iroha blocks from
```
Comment (Mingela) [inline btc-address-generation/README.md]:
```suggestion
* `btc-address-generation.irohaBlockQueue` - name of the RabbitMQ queue to read Iroha blocks from
```

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: btc-deposit/README.md [hunk 6]
```diff
@@ -1,31 +1,16 @@
...
-
-* `btc-deposit.bitcoin.blockStoragePath` - a path to Bitcoin blockchain storage.  Only headers are stored. Typically this folder consumes very little amount of disk space (approximately 50-60mb on MainNet).
-
-* `btc-deposit.bitcoin.confidenceLevel` - the minimum depth of deposit transaction in Bitcoin blockchain to be considered as available to spend.
-
-* `btc-deposit.bitcoin.hosts` - a list of Bitcoin full node hosts. Hosts are separated by a comma(`,`) symbol. These hosts are used as a source of Bitcoin blockchain. 
-
-* `btc-deposit.notaryCredential` - credentials of the Notary account. This account is used to create 'increase balance'  transactions in Iroha. Must be a  multisignature one.
+* `btc-deposit.notaryCredential` - credentials of the Notary account. This account is used to create 'increase balance' transactions in Iroha. Must be a multisignature one.
+* `btc-deposit.irohaBlockQueue` - name of a RabbitMQ queue that will be used to read Iroha blocks from
```
Comment (Mingela) [inline btc-deposit/README.md]:
```suggestion
* `btc-deposit.irohaBlockQueue` - name of the RabbitMQ queue to read Iroha blocks from
```

File: README.md [hunk 7]
```diff
@@ -1,11 +1,67 @@
...
-## How to run notary application and services in Bitcoin main net
-1) Run common services
-2) Create `.wallet` file (ask maintainers how to do that) and put it to desired location
-3) Run address generation process using `PROFILE=mainnet ./gradlew runBtcAddressGeneration`
-4) Run registration service `PROFILE=mainnet ./gradlew runBtcRegistration`
-5) Run notary service `PROFILE=mainnet ./gradlew runBtcDepositWithdrawal`
-
-## Testing Bitcoin
+## Glossary
+* BitconD - (aka full node) peer and a participant of the Bitcoin network that stores complete Bitcoin blockchain data.
```
Comment (LiraLemur) [inline README.md:4]: BitcoinD?

File: README.md [hunk 8]
```diff
@@ -1,11 +1,67 @@
...
+
+* SPV - a.k.a Simplified Payment Verification. A special mode of `BitcoinD` where only headers of Bitcoin blocks are stored.
+
+* Wallet - a special file that stores and manages private/public keys and addresses.
+
+* BitcoinJ - a Java library that we use to communicate with Bitcoin network (see https://bitcoinj.github.io/)
```
Comment (LiraLemur) [inline README.md]: not sure a is needed before special file and java library

File: README.md [hunk 9]
```diff
@@ -1,11 +1,67 @@
...
+
+* MultiSig Address - a special address in Bitcoin that requires m (threshold) of n (number of Notaries) signatures to unlock assets. 
+
+* Redeem Script - a special script that is used to describe the terms of locking/releasing coins.
```
Comment (LiraLemur) [inline README.md]: again, not sure a is necessary there

File: README.md (hunk 2 above)
Comment (LiraLemur) [reply README.md]: or make sure

File: README.md [hunk 10]
```diff
@@ -1,11 +1,67 @@
...
+
+#### Implementation details/issues 
+Address generation is a time-consuming process. It may take approximately 10-15 seconds to collect all the keys. To speed up the registration process, we generate MultiSig addresses up front.
+
+[More details](https://github.com/d3ledger/d3-btc/blob/develop/btc-address-generation/README.md)
+
+## Client Registration 
+This event will assign a Bitcoin MultiSig address to recently created Iroha account. 
+
+Client sends a registration request to particular Notary node.
```
Comment (LiraLemur) [inline README.md]: a particular?

File: README.md [hunk 11]
```diff
@@ -1,11 +1,67 @@
...
+This event will assign a Bitcoin MultiSig address to recently created Iroha account. 
+
+Client sends a registration request to particular Notary node. 
+Notary chooses MultiSig address and creates a transaction in Iroha to create a "client to Bitcoin address" relationship.  
+
+
+## Deposit Service
+Each Notary watches for transactions of MultiSig addresses assigned to our clients in the Bitcoin network. Whenever we see a transaction with coins being sent to one of the addresses, Notaries increase its balance.
+
+Iroha transfer transaction is formed as a multi-signature transaction that has two commands:` AddAssetQuantity` and `TransferAsstet`.
```
Comment (LiraLemur) [inline README.md]: extra space in ` AddAssetQuantity`

File: README.md [hunk 12]
```diff
@@ -1,11 +1,67 @@
...
+
+[More details](https://github.com/d3ledger/d3-btc/blob/develop/btc-deposit/README.md)
+
+## Withdrawal Service
+The withdrawal process works the following way: 
+
+1. D3 client sends assets to a dedicated Iroha withdrawal account.
+2. Every node reacts to this transfer by creating a Bitcoin transaction.
+3. Once a transaction is created, all the nodes try to sign recently created Bitcoin transaction if possible.
```
Comment (LiraLemur) [inline README.md]: just all nodes?

File: README.md [hunk 13]
```diff
@@ -1,11 +1,67 @@
...
+4. When enough signatures are collected(supermajority threshold), Notary forms a transaction with all the signatures and sends it to the Bitcoin network.
```
Comment (LiraLemur) [inline README.md]: ollected(supermajority threshold) - no space

File: README.md [hunk 14]
```diff
@@ -1,11 +1,67 @@
...
+[More details](https://github.com/d3ledger/d3-btc/blob/develop/btc-deposit/README.md)
+
+## Withdrawal Service
+The withdrawal process works the following way: 
+
+1. D3 client sends assets to a dedicated Iroha withdrawal account.
+2. Every node reacts to this transfer by creating a Bitcoin transaction.
+3. Once a transaction is created, all the nodes try to sign recently created Bitcoin transaction if possible.
+4. When enough signatures are collected(supermajority threshold), Notary forms a transaction with all the signatures and sends it to the Bitcoin network. 
+5. The Bitcoin withdrawal transaction is formed as a multi-input transfer with 2 outputs: destination address and change address.
```
Comment (LiraLemur) [inline README.md]: do we need The before Bitcoin withdrawal?

File: btc-address-generation/README.md [hunk 15]
```diff
@@ -1,26 +1,38 @@
 ## Bitcoin address generation service
-Bitcoin Address Generation service(or simply `btc-address-generation`) is used to create Bitcoin MultiSig addresses that may be registered by D3 clients later. The service is also responsible for creating change addresses (addresses that store changes). 
+The Bitcoin address generation service(or simply `btc-address-generation`) is used to create Bitcoin MultiSig addresses that may be registered by D3 clients later.
```
Comment (LiraLemur) [inline btc-address-generation/README.md]: service(or simply - no space, also - not sure the at the beginning is needed.

File: btc-address-generation/README.md [hunk 16]
```diff
@@ -1,26 +1,38 @@
...
- 2) Once this command appears, the service starts key pair creation and saves it in `keys.d3.wallet` file and Iroha(only public keys go to Iroha).  
-3) Then, the service waits until enough key pairs are collected in Iroha. All the Notaries must create a key pair. If enough key pairs are created, the service generates a new Bitcoin MultiSig address using public keys created by all the Notaries. This MultiSig address is also saved in `keys.d3.wallet` file and Iroha. 
+1) The service waits for a special command that triggers the address generation process. Free addresses are also generated on D3 client registration because one registration takes exactly one MultiSig address. 
+2) Once this command appears, the service starts key pair creation and saves it in `keys.d3.wallet` file and Iroha(only public keys go to Iroha).  
+3) Then, the service waits until enough public keys are collected in Iroha. Every node must create one key pair and save its public key in Iroha. If enough key pairs are created, the service generates a new Bitcoin MultiSig address using public keys created by all nodes. Then, the MultiSig address is saved in `keys.d3.wallet` file and in Iroha key-value storage.
```
Comment (LiraLemur) [inline btc-address-generation/README.md]: Then, the MultiSig address - would omit the

File: btc-address-generation/README.md [hunk 17]
```diff
@@ -1,26 +1,38 @@
...
 * `btc-address-generation.changeAddressesStorageAccount` -  Iroha account that is responsible for change addresses storage. Addresses are stored in this account details.
 * `btc-address-generation.healthCheckPort` - port of health check endpoint. A health check is available on `http://host:healthCheckPort/actuator/health`. This service checks if `btc-address-generation` is able to listen to Iroha blocks.
 * `btc-address-generation.threshold` - a number of MultiSig addresses that must be created in advance.
-* `btc-address-generation.nodeId` - identifier of Notary node. This identifier must correlate to an identifier that is set in `btc-registration` configuration file on the same node. This value must be different on different nodes. We use this identifier as a mechanism of synchronization issues resistance.
+* `btc-address-generation.nodeId` - identifier of node. This identifier must correlate to an identifier that is set in `btc-registration` configuration file on the same node. This value must be different on different nodes.
```
Comment (LiraLemur) [inline btc-address-generation/README.md]: of the node, I believe. correlate with the identifier...

File: btc-address-generation/README.md [hunk 18]
```diff
@@ -1,26 +1,38 @@
...
+* `btc-address-generation.irohaBlockQueue` - name of a RabbitMQ queue that will be used to read Iroha blocks from
+* `btc-address-generation.irohaQueryTimeoutMls` - Iroha query timeout in milliseconds. We need this value to be set in order to improve the service liveness. The service re-reads requested data from Iroha if it fails to do so from the first attempt. If the service reaches the specified timeout, it returns an error.
```
Comment (LiraLemur) [inline btc-address-generation/README.md:25]: reaches the deadline?..

File: btc-deposit/README.md [hunk 19]
```diff
@@ -1,31 +1,16 @@
-
 ## Bitcoin deposit service
-Bitcoin Deposit service(or simply `btc-deposit`) is here to listen to Bitcoin blockchain transactions and increase clients balances in Iroha blockchain. 
+The Bitcoin deposit service(or simply `btc-deposit`) is here to listen to Bitcoin blockchain transactions and increase clients balances in Iroha blockchain.
```
Comment (LiraLemur) [inline btc-deposit/README.md:2]: Would not add the

File: btc-deposit/README.md [hunk 20]
```diff
@@ -1,31 +1,16 @@
...
-2) Then it starts Bitcoin blockchain downloading process (only headers are stored on the disk)
-3) After that, the service increases client's balance in Iroha blockchain if needed. This 'increase' works in a multisignature fashion. So every node must create the same Iroha transaction. Bitcoin block time is used as a source of time for the Iroha 'increase balance' transaction. Doing that we can guarantee that every node will use the same time.
+1) The service creates Bitcoin blockchain listeners that listen to transactions where coins were sent to our clients. The listener waits until the transaction of interest reaches at least 6(or another amount of blocks which may be specified) blocks in depth. Once it happens, an Iroha 'increase balance' transaction is created. This 'increase' works in a multisignature fashion. So every node must create the same Iroha transaction. Bitcoin block time is used as a source of time for the Iroha 'increase balance' transaction.
```
Comment (LiraLemur) [inline btc-deposit/README.md]: waits for the transaction of interest to reach, least 6(or another – no space, an Iroha  - definitely no an, an Iroha

File: btc-dw-bridge/README.md [hunk 21]
```diff
@@ -0,0 +1,23 @@
+## Bitcoin deposit-withdrawal bridge
+The Bitcoin deposit-withdrawal bridge(or simply `btc-dw-bridge`) is module that combines the Bitcoin deposit and the Bitcoin withdrawal services in a single deployment unit.
```
Comment (LiraLemur) [inline btc-dw-bridge/README.md]: is a module, I really don't think 'the's are necessary here.

File: btc-dw-bridge/README.md [hunk 22]
```diff
@@ -0,0 +1,23 @@
+## Bitcoin deposit-withdrawal bridge
+The Bitcoin deposit-withdrawal bridge(or simply `btc-dw-bridge`) is module that combines the Bitcoin deposit and the Bitcoin withdrawal services in a single deployment unit. 
+We have to run these service together due several technical issues and bitcoinj library capabilities.
```
Comment (LiraLemur) [inline btc-dw-bridge/README.md]: due to

File: btc-dw-bridge/README.md [hunk 23]
```diff
@@ -0,0 +1,23 @@
+## Bitcoin deposit-withdrawal bridge
+The Bitcoin deposit-withdrawal bridge(or simply `btc-dw-bridge`) is module that combines the Bitcoin deposit and the Bitcoin withdrawal services in a single deployment unit. 
+We have to run these service together due several technical issues and bitcoinj library capabilities.  
+
+### Configuration overview (dw-bridge.properties)
+* `btc-dw-bridge.bitcoin.blockStoragePath` - a path to Bitcoin blockchain storage.  Only headers are stored. Typically this folder takes a little amount of disk space (approximately 50-60mb on MainNet).
```
Comment (LiraLemur) [inline btc-dw-bridge/README.md]: e.  On – extra space, Typically, - needs comma,

File: btc-dw-bridge/README.md [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+* `btc-dw-bridge.bitcoin.confidenceLevel` - the minimum depth of deposit transaction in Bitcoin blockchain to be considered as available to spend.
+* `btc-dw-bridge.bitcoin.hosts` - a list of Bitcoin full node hosts. Hosts are separated by a comma(`,`) symbol. These hosts are used as a source of Bitcoin blockchain. 
+* `btc-dw-bridge.dnsSeedAddresses` - a list of Bitcoin DNS seeds. These addresses are used to discover Bitcoin full nodes. Seeds are separated by a comma(`,`) symbol.
+* `btc-dw-bridge.minBlockHeightForPeer` - minimum amount of blocks for a connected Bitcoin node to have. If the connected node has less blocks than specified, it's disconnected.
+* `btc-dw-bridge.irohaQueryTimeoutMls` - Iroha query timeout in milliseconds. We need this value to be set in order to improve the service liveness. The service re-reads requested data from Iroha if it fails to do so from the first attempt. If the service reaches the specified timeout, it returns an error.
```
Comment (LiraLemur) [inline btc-dw-bridge/README.md:11]: again - not sure about reaching the timeout

File: btc-withdrawal/README.md [hunk 25]
```diff
@@ -1,33 +1,26 @@
 ## Bitcoin withdrawal service
-Bitcoin Withdrawal service(or simply `btc-withdrawal`) is a service dedicated to creating and signing Bitcoin withdrawal transactions.
+The Bitcoin withdrawal service(or simply `btc-withdrawal`) is a service dedicated to creating and signing Bitcoin withdrawal transactions.
```
Comment (LiraLemur) [inline btc-withdrawal/README.md:2]: again – the

File: btc-withdrawal/README.md [hunk 26]
```diff
@@ -1,33 +1,26 @@
...
 
 ### Simplified flow
-1) The service listens to transfer commands in Iroha blockchain using `chain-adapter` in order to achieve high stability while handling blocks.
-2) If a transfer command is valid (destination address is whitelisted and has base58 format), then the service tries to establish 'withdrawal consensus' among all the Notaries. They must decide what UTXO may be spent according to the Bitcoin blockchain state.
+1) The service listens to transfer commands in Iroha blockchain
+2) If a transfer command is valid (destination address is whitelisted and it has a `base58` format), then the service tries to establish 'withdrawal consensus' among all the nodes. They must decide what UTXO may be spent.
```
Comment (LiraLemur) [inline btc-withdrawal/README.md]: all nodes or all of the nodes

File: btc-withdrawal/README.md [hunk 27]
```diff
@@ -1,33 +1,26 @@
...
 3) Once the 'withdrawal consensus' is reached, the service starts creating a Bitcoin withdrawal transaction.
-4) The next step is signing. All the Notaries must sign a newly created transaction.
-5) If all the Notaries signed a Bitcoin withdrawal transaction properly, it will be sent to the Bitcoin network.
+4) The next step is signing. All the nodes must sign a newly created transaction.
```
Comment (LiraLemur) [inline btc-withdrawal/README.md]: all or all of the nodes

File: btc-withdrawal/README.md [hunk 28]
```diff
@@ -1,33 +1,26 @@
...
+5) If all the nodes(or the majority of them) signed a Bitcoin withdrawal transaction properly, it will be sent to the Bitcoin network.
```
Comment (LiraLemur) [inline btc-withdrawal/README.md]: odes(or  - no space

**Review by LiraLemur — APPROVED** [review]

**Pull request merged**